controllers/certificaterequest/certificaterequest_controller.go

```diff
@@ -111,24 +111,26 @@ func (r *CertificateRequestReconciler) Reconcile(ctx context.Context, request re
 	err := r.Client.Get(context.TODO(), request.NamespacedName, cr)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			// Request object not found, could have been deleted after reconcile request.
-			// Update metrics to show it's missing and set CertValidDuration to 0
-			localmetrics.UpdateCertValidDuration(r.Client, nil, time.Now(), request.Namespace, request.Namespace)
 			return reconcile.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
 		reqLogger.Error(err, err.Error())
-		localmetrics.UpdateCertValidDuration(r.Client, nil, time.Now(), cr.Namespace, cr.Namespace)
 		return reconcile.Result{}, err
 	}
 
 	// Handle the presence of a deletion timestamp.
 	if !cr.DeletionTimestamp.IsZero() {
-		// Set CertValidDuration to 0 for certificates being deleted
-		localmetrics.UpdateCertValidDuration(r.Client, nil, time.Now(), cr.Namespace, cr.Namespace)
```

Comment:
Does there not need to be any change to the cert metrics if it's deleted?  Or is this cleaned up?

Reply:
In hindsight, this would open us up to an edge-case where a certificate gets close to it's expiry, triggering an alert, but the cluster gets deleted before certman has a chance to renew it. 

In such a case, the metrics would be "locked" at the close-to-expiry value, and the alert would therefore never clear on it's own.

Reply:
It looks like prometheus provides a [DeletePartialMatch](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#MetricVec.DeletePartialMatch) method that allows us to clear these metrics for a specific certificaterequest without having to retrieve the certificate secret (and, therefore, avoiding the potential for a race-condition with hive as cluster objects get cleaned up)